### PR TITLE
feat: export route table from environment stack

### PR DESCRIPTION
**Pull request by Lou1415926**
This PR addresses #2660. If an environment uses Copilot-default VPC, then the public route table ID is exported; as a result, user can import the route table ID in their addon stack by writing `'Fn::ImportValue': !Sub '${App}-${Env}-PublicRouteTableID`.

Note that the route table is only exported if Copilot-default VPC is used. If the user chooses to import their own VPC, Copilot doesn't export the route table.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

**Review by huanjani — APPROVED** [review]
Thank you!

**Comment by efekarakus** [comment]
Do we need to bump the version number of the environment template in order for these changes to be picked up?

**Comment by gautam-nutalapati** [comment]
Probably we should also add check in [integration test](https://github.com/aws/copilot-cli/blob/09ae32389f000258666f7055d8a5afa5581874cc/internal/pkg/deploy/cloudformation/cloudformation_integration_test.go#L525) to validate this export.

File: internal/pkg/deploy/env.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ const (
 	// LegacyEnvTemplateVersion is the version associated with the environment template before we started versioning.
 	LegacyEnvTemplateVersion = "v0.0.0"
 	// LatestEnvTemplateVersion is the latest version number available for environment templates.
-	LatestEnvTemplateVersion = "v1.6.1"
+	LatestEnvTemplateVersion = "v1.6.2"
```
Comment (efekarakus) [inline internal/pkg/deploy/env.go]: nit: Technically it's a feature and not a bug fix :P 
```suggestion
	LatestEnvTemplateVersion = "v1.7.0"
```

**Review by efekarakus — APPROVED** [review]

**Pull request merged**